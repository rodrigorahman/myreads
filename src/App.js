import React from 'react'
import './App.css'
import "react-progress-2/main.css"

import Search from "./screens/bookcase/Search";
import Header from "./screens/header/Header";
import Bookcase from "./screens/bookcase/Bookcase";
import FloatButton from "./shared/FloatButton";
import { Route } from "react-router-dom";


class BooksApp extends React.Component {

    state = {
        books: [],
    };

    callbackUpdateState = () => {
        const books = this.state.books.filter((b) => b.status !== 'none');
        this.setState({books});
        this.saveBooksInLocalstorage();
    };

    saveBooksInLocalstorage = () => {
        localStorage.books = JSON.stringify(this.state.books);
    };


    readBooksLocalStorage = () => {
        let booksLocal = localStorage.books;
        let books = [];

        if (booksLocal) {
            books = JSON.parse(booksLocal);
        }

        return books;
    };

    componentWillMount() {
        this.setState({books: this.readBooksLocalStorage()});
    }

    render() {
        return (
            <div className="app">
                <Route
                    exact
                    path='/search'
                    render={({history}) => (
                        <Search
                            books={this.state.books}
                            callbackUpdateState={this.callbackUpdateState}
                            history={history}
                        />
                    )}
                />

                <Route
                    exact
                    path='/'
                    render={({history}) => (

                        <div className="list-books">

                            <Header
                                books={this.state.books}
                                callbackUpdateState={this.callbackUpdateState}/>

                            <Bookcase
                                books={this.state.books}
                                callbackUpdateState={this.callbackUpdateState}
                                filterByCategory={true}
                                history={history}
                            />

                            <FloatButton onClick={() => history.push('/search')}/>
                        </div>
                    )}
                />

            </div>
        )
    }
}

export default BooksApp