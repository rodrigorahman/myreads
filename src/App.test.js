import React from 'react'
import App from './App'

import { shallow } from 'enzyme'
import 'core-js/es6/map'
import 'core-js/es6/set'

localStorage.books = JSON.stringify([{}])

it('renders without crashing', () => {
  expect(shallow(
    <App/>
  )).toMatchSnapshot()
})