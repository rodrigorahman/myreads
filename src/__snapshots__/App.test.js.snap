// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="app"
>
  <Route
    exact={true}
    path="/search"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
</div>
`;
